src/Version.h

```diff
@@ -17,26 +17,79 @@
 
 namespace Version {
 
-using exact_width_mumble_raw_version_t = std::uint32_t;
-using mumble_raw_version_t             = unsigned int;
+enum CompareMode { AtLeast, LessThan };
+
+//
+// The mumble version format (v2) is a uint64 and consists of the following:
+// reserved/unused  major   minor   patch
+// 0xFFFF           0xFFFF  0xFFFF  0xFFFF
+// (big-endian)
+//
+
+const std::uint64_t OFFSET_UNUSED = 48;
+const std::uint64_t OFFSET_MAJOR  = 32;
+const std::uint64_t OFFSET_MINOR  = 16;
+const std::uint64_t OFFSET_PATCH  = 0;
+const std::uint64_t FIELD_WIDTH   = 0xFFFF;
+const std::uint64_t FIELD_UNUSED  = (FIELD_WIDTH << Version::OFFSET_UNUSED);
+const std::uint64_t FIELD_MAJOR   = (FIELD_WIDTH << Version::OFFSET_MAJOR);
+const std::uint64_t FIELD_MINOR   = (FIELD_WIDTH << Version::OFFSET_MINOR);
+const std::uint64_t FIELD_PATCH   = (FIELD_WIDTH << Version::OFFSET_PATCH);
+
+using exact_width_mumble_raw_version_t = std::uint64_t;
+using mumble_raw_version_t             = unsigned long int;
 
 static_assert(sizeof(exact_width_mumble_raw_version_t) <= sizeof(mumble_raw_version_t),
```

Comment:
@Krzmbrzl What is the reason for this? Why not make mumble_raw_version a std::unit64_t and call it a day?

Reply:
I was worried that on some architectures the performance might be better by using a larger integer than requiring strictly 64-bit width (in contrast to at-least 64-bit width). However, in hindsight the differences will probably not even be measurable, so I'd be fine with just defining the raw_version_t to be `std::uint64_t`.